pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/api.go

```diff
@@ -266,32 +292,51 @@ func (s *apiScaffolder) updateControllerCode(controller controllers.Controller)
 	return nil
 }
 
-func (s *apiScaffolder) scaffoldCreateAPIFromKustomize() error {
+func (s *apiScaffolder) scaffoldCreateAPIFromKustomize(isLegacyLayout bool) error {
 	// Now we need call the kustomize/v1 plugin to do its scaffolds when we create a new API
 	// todo: when we have the go/v4-alpha plugin we will also need to check what is the plugin used
 	// in the Project layout to know if we should use kustomize/v1 OR kustomize/v2-alpha
-	kustomizeV1Scaffolder := kustomizev1scaffolds.NewAPIScaffolder(
-		s.config,
-		s.resource,
-		true,
-	)
-	kustomizeV1Scaffolder.InjectFS(s.fs)
-
-	if err := kustomizeV1Scaffolder.Scaffold(); err != nil {
-		return fmt.Errorf("error scaffolding kustomize files for the APIs: %v", err)
+	if isLegacyLayout {
+		kustomizeV1Scaffolder := kustomizev1scaffolds.NewAPIScaffolder(
+			s.config,
+			s.resource,
+			true,
+		)
+		kustomizeV1Scaffolder.InjectFS(s.fs)
+
+		if err := kustomizeV1Scaffolder.Scaffold(); err != nil {
+			return fmt.Errorf("error scaffolding kustomize files for the APIs: %v", err)
+		}
+	} else {
+		kustomizeV2Scaffolder := kustomizev2scaffolds.NewAPIScaffolder(
+			s.config,
+			s.resource,
+			true,
+		)
+		kustomizeV2Scaffolder.InjectFS(s.fs)
+
+		if err := kustomizeV2Scaffolder.Scaffold(); err != nil {
+			return fmt.Errorf("error scaffolding kustomize files for the APIs: %v", err)
+		}
```

Comment:
minor nit for reducing code duplication since both `NewAPIScaffolder()` functions return the same type:
```go
var kustomizeScaffolder plugins.Scaffolder

if isLegacyLayout {
  kustomizeScaffolder = kustomizev1scaffolds.NewAPIScaffolder(
	s.config,
	s.resource,
	true,
  )
} else {
  kustomizeScaffolder = kustomizev2scaffolds.NewAPIScaffolder(
	s.config,
	s.resource,
	true,
  )
}

kustomizeScaffolder.InjectFS(s.fs)

if err := kustomizeScaffolder.Scaffold(); err != nil {
	return fmt.Errorf("error scaffolding kustomize files for the APIs: %v", err)
}
```